packages/core/test/actions.test.ts

```diff
@@ -17,6 +19,49 @@ import {
   send
 } from '../src/actions';
 
+const seen = new WeakSet<AnyStateMachine>();
+
+const trackEntries = <TMachine extends AnyStateMachine>(machine: TMachine) => {
```

Comment:
This could use a better name, and maybe even a better API. Some of the APIs I had in mind initially were harder to type so I've just chosen something that I could make very quickly.